projects/ngx-d3timeline/src/lib/content/selectors/activity.selectors.ts

```diff
@@ -46,8 +53,39 @@ const selectDraggedToSeries = createSelector(
   valueToSeries
 );
 
+const selectDeltaTime = createSelector(
+  selectTimeOrientation,
+  selectDragEvent,
+  getDeltaTime
+);
+
+const selectCurrentlyDraggedUpdatedStart = createSelector(
+  selectCurrentlyDraggedActivity,
+  activity => activity && activity.updatedStart
```

Comment:
why do these need to return the activity? Tried to trace it through to where it's used and it looks like just the time is used?

Reply:
Its used in `selectActivityUpdatedForDrag`

Reply:
In the code with the renamings it is used as an input to `selectShiftedByDeltaTime` where it is passed as a selector `selectTime`, the result of which is passed into the first parameter of `shiftTimeByRangeValue`. Note this parameter is a Date/time only, it does not use the Activity which comes back to my original question.

Reply:
Sorry I don't quite understand

> why do these need to return the activity? Tried to trace it through to where it's used and it looks like just the time is used?

They don't return the activity, they return a time value from an activity i.e. updatedStart or updatedEnd.

Reply:
ah yes I see now, I was confusing `activity && activity.updatedStart` as a union type when its just a logical operator.